homeassistant/components/sia/config_flow.py

```diff
@@ -0,0 +1,161 @@
+"""Config flow for sia integration."""
+from __future__ import annotations
+
+import logging
+
+from pysiaalarm import (
+    InvalidAccountFormatError,
+    InvalidAccountLengthError,
+    InvalidKeyFormatError,
+    InvalidKeyLengthError,
+    SIAAccount,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_PORT, CONF_PROTOCOL
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_ACCOUNT,
+    CONF_ACCOUNTS,
+    CONF_ADDITIONAL_ACCOUNTS,
+    CONF_ENCRYPTION_KEY,
+    CONF_IGNORE_TIMESTAMPS,
+    CONF_PING_INTERVAL,
+    CONF_ZONES,
+    DOMAIN,
+    TITLE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+HUB_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_PORT): int,
+        vol.Optional(CONF_PROTOCOL, default="TCP"): vol.In(["TCP", "UDP"]),
+        vol.Required(CONF_ACCOUNT): str,
+        vol.Optional(CONF_ENCRYPTION_KEY): str,
+        vol.Required(CONF_PING_INTERVAL, default=1): int,
+        vol.Required(CONF_ZONES, default=1): int,
+        vol.Required(CONF_IGNORE_TIMESTAMPS, default=False): bool,
+        vol.Optional(CONF_ADDITIONAL_ACCOUNTS, default=False): bool,
+    }
+)
+
+ACCOUNT_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ACCOUNT): str,
+        vol.Optional(CONF_ENCRYPTION_KEY): str,
+        vol.Required(CONF_PING_INTERVAL, default=1): int,
+        vol.Required(CONF_ZONES, default=1): int,
+        vol.Required(CONF_IGNORE_TIMESTAMPS, default=False): bool,
+        vol.Optional(CONF_ADDITIONAL_ACCOUNTS, default=False): bool,
+    }
+)
+
+
+def validate_input(data: ConfigType):
+    """Validate the input by the user."""
+    SIAAccount.validate_account(data[CONF_ACCOUNT], data.get(CONF_ENCRYPTION_KEY))
+
+    try:
+        ping = int(data[CONF_PING_INTERVAL])
+        assert 1 <= ping <= 1440
+    except AssertionError as invalid_ping:
+        raise InvalidPing from invalid_ping
+    try:
+        zones = int(data[CONF_ZONES])
+        assert zones > 0
+    except AssertionError as invalid_zone:
+        raise InvalidZones from invalid_zone
+
+
+class SIAConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for sia."""
+
+    VERSION: int = 1
+    CONNECTION_CLASS: str = config_entries.CONN_CLASS_LOCAL_PUSH
+    data: ConfigType = {}
+
+    async def async_step_add_account(self, user_input: ConfigType = None):
+        """Handle the additional accounts steps."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user",
```

Comment:
We should only display the form for each step when we move between steps.

If you want to to back to user, please `await self.async_step_user()`